app/js/components/labOrderEntry/LabEntryForm.jsx

```diff
@@ -67,10 +123,7 @@ export class LabEntryForm extends React.Component {
   }
 
   handleSubmit = () => {
-    /* *
-     * this should contain data to be submitted
-     * * */
-    this.handleCancel();
+    this.props.createLabOrder(this.state);
```

Comment:
I see you are passing the whole `state` of this component to the `createLabOrder` action. I don't think this is a good idea.
From the implementation of the `createLabOrder` action you created, you stringify the argument it receives and save it straight away into the database. 
Calling the `createLabOrder` action like this means you would end up stringifying the whole `state` of this component and save it into the database.
The `state` of this component contains a lot that are not even part of the lab order information, I think you need to do some refactoring here.

Reply:
ok, I'll look into that